edb/edgeql-parser/edgeql-parser-python/src/pynormalize.rs

```diff
@@ -98,12 +106,80 @@ impl Entry {
     }
 }
 
-pub fn serialize_extra(variables: &[Variable]) -> Result<Bytes, String> {
+struct VariableNameIter<'a> {
+    entry_pack: &'a PackedEntry,
+    first_extra: Option<usize>,
+    extra_named: bool,
+    name_index: usize,
+    blob_index: usize,
+    var_index: usize,
+}
+
+impl<'a> VariableNameIter<'a> {
+    pub fn new(entry: &'a Entry) -> Self {
+        VariableNameIter {
+            entry_pack: &entry.entry_pack,
+            first_extra: entry.first_extra,
+            extra_named: entry.extra_named,
+            name_index: 0,
+            blob_index: 0,
+            var_index: 0,
+        }
+    }
+}
+
+impl<'a> Iterator for VariableNameIter<'a> {
+    type Item = (String, usize, usize, &'a Value);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        // Check termination
+        let first = match self.first_extra {
+            Some(first) => first,
+            None => return None,
+        };
```

Comment:
More concisely written as `let first = self.first_extra?;`, I believe. This will use the Rust try operation to early exit. I believe clippy might recommend this to you.